kratos/sources/integration_rules.cpp

```diff
@@ -29,1394 +29,161 @@
-        IntegrationPointType(-0.4058451513773972, 0.3818300505051189),
-        IntegrationPointType( 0.0000000000000000, 0.4179591836734694),
-        IntegrationPointType( 0.4058451513773972, 0.3818300505051189),
-        IntegrationPointType( 0.7415311855993945, 0.2797053914892766),
-        IntegrationPointType( 0.9491079123427585, 0.1294849661688697)
-    }
-};
-
-LineGaussLegendreIntegrationPoints8::IntegrationPointsArrayType LineGaussLegendreIntegrationPoints8::msIntegrationPoints =
-{
-    {
-        IntegrationPointType(-0.9602898564975363, 0.1012285362903763),
-        IntegrationPointType(-0.7966664774136267, 0.2223810344533745),
-        IntegrationPointType(-0.5255324099163290, 0.3137066458778873),
-        IntegrationPointType(-0.1834346424956498, 0.3626837833783620),
-        IntegrationPointType( 0.1834346424956498, 0.3626837833783620),
-        IntegrationPointType( 0.5255324099163290, 0.3137066458778873),
-        IntegrationPointType( 0.7966664774136267, 0.2223810344533745),
-        IntegrationPointType( 0.9602898564975363, 0.1012285362903763)
-    }
-};
-
-LineGaussLegendreIntegrationPoints9::IntegrationPointsArrayType LineGaussLegendreIntegrationPoints9::msIntegrationPoints =
-{
-    {
-        IntegrationPointType(-0.9681602395076261, 0.0812743883615744),
-        IntegrationPointType(-0.8360311073266358, 0.1806481606948574),
-        IntegrationPointType(-0.6133714327005904, 0.2606106964029354),
-        IntegrationPointType(-0.3242534234038089, 0.3123470770400029),
-        IntegrationPointType( 0.0000000000000000, 0.3302393550012598),
-        IntegrationPointType( 0.3242534234038089, 0.3123470770400029),
-        IntegrationPointType( 0.6133714327005904, 0.2606106964029354),
-        IntegrationPointType( 0.8360311073266358, 0.1806481606948574),
-        IntegrationPointType( 0.9681602395076261, 0.0812743883615744)
-    }
-};
-
-LineGaussLegendreIntegrationPoints10::IntegrationPointsArrayType LineGaussLegendreIntegrationPoints10::msIntegrationPoints =
-{
-    {
-        IntegrationPointType(-0.9739065285171717, 0.0666713443086881),
-        IntegrationPointType(-0.8650633666889845, 0.1494513491505806),
-        IntegrationPointType(-0.6794095682990244, 0.2190863625159820),
-        IntegrationPointType(-0.4333953941292472, 0.2692667193099963),
-        IntegrationPointType(-0.1488743389816312, 0.2955242247147529),
-        IntegrationPointType( 0.1488743389816312, 0.2955242247147529),
-        IntegrationPointType( 0.4333953941292472, 0.2692667193099963),
-        IntegrationPointType( 0.6794095682990244, 0.2190863625159820),
-        IntegrationPointType( 0.8650633666889845, 0.1494513491505806),
-        IntegrationPointType( 0.9739065285171717, 0.0666713443086881)
-    }
-};
+LineGaussLegendreIntegrationPoints1::IntegrationPointsArrayType LineGaussLegendreIntegrationPoints1::msIntegrationPoints = LineGaussLegendreIntegrationPoints1::IntegrationPoints();

```

Comment:
would 'auto' not work for all of these?

Reply:
Nope, I am accessing an static variable. Any way, if you can avoid auto, avoid it. (some cases can be wrong)

Reply:
I see! Thanks